parlai/utils/bpe.py

```diff
@@ -219,6 +248,18 @@ def sync_with_dict(self, dict_agent):
             agent with which we are syncing the dictionary
         """
 
+    def add_special_tokens(self, dict_agent, special_tokens: List[str]):
+        """
+        Add special tokens to the tokenizer.
+
+        These tokens are treated as discreet tokens, and prioritized over the BPE
+        tokenization.
+        """
+        # note, HF ByteLevelBPE tokenizer handles special tokens itself in
+        # a special way
+        for token in special_tokens:
+            self._special_tokens[token] = 1
```

Comment:
hmm why a dict and not a set, if the val is always 1? Do we need a fixed sort order?

Reply:
I'm exploiting the fact that dictionaries in python 3.5+ are insert-ordered to emulate an ordered set. sets are unordered.